--- widelands-build19-src/CMakeLists.txt~	2019-04-06 09:55:34.000000000 +0200
+++ widelands-build19-src/CMakeLists.txt	2019-04-06 10:02:33.509371888 +0200
@@ -183,7 +183,6 @@
 wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=format-security")
 wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=return-type")
 wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=shadow")
-wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror=uninitialized")
 
 IF (WIN32)
   add_definitions(-DMINGW_HAS_SECURE_API)
