--- widelands-1.2/CMakeLists.txt~	2024-03-28 10:12:05.000000000 +0000
+++ widelands-1.2/CMakeLists.txt	2024-03-30 10:51:23.584411957 +0000
@@ -200,6 +200,10 @@
   option(OPTION_ASAN "Build with AddressSanitizer" OFF)
 elseif(CMAKE_BUILD_TYPE STREQUAL "None")
   message(STATUS "Not setting any default flags.")
+elseif(CMAKE_BUILD_TYPE STREQUAL "PLD")
+  set(WL_OPTIMIZE_FLAGS "")
+  set(WL_DEBUG_FLAGS "-DNDEBUG -DNOPARACHUTE")
+  option(OPTION_ASAN "Build with AddressSanitizer" OFF)
 else()
   message(FATAL_ERROR "Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
@@ -328,7 +332,7 @@
 endif()
 
 if(NOT MSVC)
-  if(CMAKE_BUILD_TYPE STREQUAL "Release")
+  if( (CMAKE_BUILD_TYPE STREQUAL "PLD") OR (CMAKE_BUILD_TYPE STREQUAL "Release") )
     message(STATUS "Compiler warnings will be ignored.")
   elseif(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (
          ((NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)) AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.2))
@@ -482,7 +486,6 @@
 install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/VERSION
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreVersionFile
 )
 
@@ -498,7 +501,6 @@
   FILES
     data/datadirversion
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT VersionFile
 )
 
@@ -515,7 +517,6 @@
     data/txts
     data/world
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreDataFiles
 )
 
@@ -523,7 +524,6 @@
   DIRECTORY
     data/maps
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT MapFiles
 )
 
@@ -532,7 +532,6 @@
     data/music
     data/sound
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT MusicFiles
 )
 
@@ -542,7 +541,6 @@
     CREDITS
     ChangeLog
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release;None
   COMPONENT CoreLicenseFiles
 )
 
@@ -561,7 +559,6 @@
     DIRECTORY
       ${CMAKE_CURRENT_BINARY_DIR}/locale/
     DESTINATION ${WL_INSTALL_DATADIR}/locale
-    CONFIGURATIONS Debug;Release;None
     COMPONENT CoreLanguageFiles
   )
 endif(OPTION_BUILD_TRANSLATIONS)
