--- widelands-build19-src/src/graphic/text/bidi.h.orig	2016-11-10 08:28:01.000000000 +0100
+++ widelands-build19-src/src/graphic/text/bidi.h	2019-04-13 14:11:05.664106041 +0200
@@ -40,7 +40,7 @@
 // True if a string contains a character from the script's code blocks
 bool has_script_character(const char* input, UI::FontSets::Selector script);
 UI::FontSet const* find_fontset(const char* word, const UI::FontSets& fontsets);
-std::string icustring2string(const UnicodeString& convertme);
+std::string icustring2string(const icu::UnicodeString& convertme);
 std::string icuchar2string(const UChar& convertme);
 bool cannot_start_line(const UChar& c);
 bool cannot_end_line(const UChar& c);
