--- widelands-1.3/CMakeLists.txt~	2025-12-18 18:19:33.000000000 +0000
+++ widelands-1.3/CMakeLists.txt	2025-12-21 13:46:53.328184238 +0000
@@ -214,6 +214,10 @@
   option(OPTION_ASAN "Build with AddressSanitizer" OFF)
 elseif(CMAKE_BUILD_TYPE STREQUAL "None")
   message(STATUS "Not setting any default flags.")
+elseif(CMAKE_BUILD_TYPE STREQUAL "PLD")
+  set(WL_OPTIMIZE_FLAGS "")
+  set(WL_DEBUG_FLAGS "-DNDEBUG -DNOPARACHUTE")
+  option(OPTION_ASAN "Build with AddressSanitizer" OFF)
 else()
   message(FATAL_ERROR "Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
@@ -363,7 +367,7 @@
 endif()
 
 if(NOT MSVC)
-  if((CMAKE_BUILD_TYPE STREQUAL "Release") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
+  if((CMAKE_BUILD_TYPE STREQUAL "PLD") OR (CMAKE_BUILD_TYPE STREQUAL "Release") OR (CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
     message(STATUS "Compiler warnings will be ignored.")
   elseif(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (
          ((NOT (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.0)) AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 12.2))
@@ -522,7 +526,6 @@
 install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/VERSION
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT CoreVersionFile
 )
 
@@ -545,7 +548,6 @@
   FILES
     data/datadirversion
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT VersionFile
 )
 
@@ -562,7 +564,6 @@
     data/txts
     data/world
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT CoreDataFiles
 )
 
@@ -570,7 +571,6 @@
   DIRECTORY
     data/maps
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT MapFiles
 )
 
@@ -579,7 +579,6 @@
     data/music
     data/sound
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT MusicFiles
 )
 
@@ -589,7 +588,6 @@
     CREDITS
     ChangeLog
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release;RelWithDebInfo;None
   COMPONENT CoreLicenseFiles
 )
 
@@ -597,7 +595,6 @@
   DIRECTORY
     doc
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug
   COMPONENT DocFiles
   PATTERN "CMakeLists.txt" EXCLUDE
 )
