--- widelands-build19-src/CMakeLists.txt.orig	2016-11-10 08:28:01.000000000 +0100
+++ widelands-build19-src/CMakeLists.txt	2017-07-15 17:41:44.154500573 +0200
@@ -98,6 +98,9 @@
 elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
   set(WL_OPTIMIZE_FLAGS "-O3")
   set(WL_DEBUG_FLAGS "-g -DNDEBUG -DNOPARACHUTE")
+elseif(CMAKE_BUILD_TYPE STREQUAL "PLD")
+  set(WL_OPTIMIZE_FLAGS "")
+  set(WL_DEBUG_FLAGS "-DNDEBUG -DNOPARACHUTE")
 else()
   message(FATAL_ERROR "Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}")
 endif()
--- widelands-build19-src/CMakeLists.txt.orig	2017-07-15 17:44:06.607832280 +0200
+++ widelands-build19-src/CMakeLists.txt	2017-07-15 18:59:17.074447445 +0200
@@ -248,7 +248,6 @@
 install (
   FILES ${CMAKE_CURRENT_BINARY_DIR}/VERSION
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release
   COMPONENT CoreVersionFile
 )
 
@@ -271,7 +270,6 @@
     data/txts
     data/world
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
   COMPONENT CoreDataFiles
 )
 
@@ -279,7 +277,6 @@
   DIRECTORY
     data/maps
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
   COMPONENT MapFiles
 )
 
@@ -288,7 +285,6 @@
     data/music
     data/sound
   DESTINATION ${WL_INSTALL_DATADIR}
-  CONFIGURATIONS Debug;Release
   COMPONENT MusicFiles
 )
 
@@ -298,7 +294,6 @@
     CREDITS
     ChangeLog
   DESTINATION ${WL_INSTALL_BASEDIR}
-  CONFIGURATIONS Debug;Release
   COMPONENT CoreLicenseFiles
 )
 
@@ -317,6 +312,5 @@
   DIRECTORY
     ${CMAKE_CURRENT_BINARY_DIR}/locale/
   DESTINATION ${WL_INSTALL_DATADIR}/locale
-  CONFIGURATIONS Debug;Release
   COMPONENT CoreLanguageFiles
 )
--- widelands-build19-src/cmake/WlFunctions.cmake.orig	2016-11-10 08:28:01.000000000 +0100
+++ widelands-build19-src/cmake/WlFunctions.cmake	2017-07-15 20:20:09.604392035 +0200
@@ -276,5 +276,5 @@
 
   #Quoting the CMake documentation on DESTINATION:
   #"If a relative path is given it is interpreted relative to the value of CMAKE_INSTALL_PREFIX"
-  install(TARGETS ${NAME} DESTINATION "." COMPONENT ExecutableFiles)
+  install(TARGETS ${NAME} DESTINATION "bin" COMPONENT ExecutableFiles)
 endfunction()
